lib/cib/cib_attrs.c

```diff
@@ -86,10 +93,12 @@ find_nvpair_attr_delegate(cib_t * the_cib, const char *attr, const char *section
     }
 
     xpath_string = calloc(1, xpath_max);
-    offset += snprintf(xpath_string + offset, xpath_max - offset, "%.128s", get_object_path(section));
```

Comment:
I would have thought that any attribute name over 128 would have been truncated and nothing would be written if `xpath_max - offset < 0`.

It seems `snprintf()` is unconditionally writing `\0` :-(